### PR TITLE
APPS-1124 revert PR326

### PR DESCRIPTION
**Pull request by YuxinShi0423**

### CODE REVIEW

File: compiler/src/main/scala/dx/translator/cwl/ProcessTranslator.scala [hunk 1]
```diff
@@ -575,24 +575,14 @@ case class ProcessTranslator(cwlBundle: CwlBundle,
         param.name -> param.dxType
       }.toMap
 
-      val (instanceType, container, attributes, requirements) = availableDependencies
-        .collectFirst({
-          case (iname, i: Application) if iname == calleeName.getOrElse("") => {
-            (i.instanceType, i.container, i.attributes, i.requirements)
-          }
-        })
-        .getOrElse((DefaultInstanceType, NoImage, Vector.empty, Vector.empty))
```
Comment (YuxinShi0423) [inline compiler/src/main/scala/dx/translator/cwl/ProcessTranslator.scala:584]: PR#326 only covers the case in which the `callee` in cwl or the `innerCall` in wdl is an application. If the inner callable of a frag applet is a subworkflow (which does not have `container` attr), the frag applet would not be given network access during compilation. 

So here I would just revert this change, and in `ApplicationCompiler` the frag applet will still be given full access like before.

File: compiler/src/main/scala/dx/compiler/ApplicationCompiler.scala [hunk 2]
```diff
@@ -448,6 +440,8 @@ case class ApplicationCompiler(typeAliases: Map[String, Type],
       case ExecutableKindWorkflowOutputReorg =>
         // The reorg applet requires higher permissions to organize the output directory.
         Some(DxAccess.empty.copy(project = Some(DxAccessLevel.Contribute)))
+      case ExecutableKindWfFragment(_, _, _, _) =>
+        Some(DxAccess.empty.copy(network = Some(Vector("*"))))
```
Comment (YuxinShi0423) [inline compiler/src/main/scala/dx/compiler/ApplicationCompiler.scala:444]: Frag applets have full access, while the other kinds `ExecutableKindWfInputs / ExecutableKindWfOutputs / ExecutableKindWfCustomReorgOutputs` won't.

These kinds of applets are inputs or outputs (default or custom reorg) helpers, so my understanding is that they do not need extra access permission.

**Comment by YuxinShi0423** [comment]
Tests that failed in release tests now passed:
`globbingscatter`: https://staging.dnanexus.com/projects/Fy9QqgQ0yzZbg9KXKP4Jz6Yq/monitor/analysis/G9PgF980yzZkj2fb5QV2J397
`public_http_import`:
https://staging.dnanexus.com/projects/Fy9QqgQ0yzZbg9KXKP4Jz6Yq/monitor/analysis/G9PgF780yzZq11Zp5Fk0832K

Similar cwl test:
`cwl_three_step_caller_wf`:
https://staging.dnanexus.com/projects/Fy9QqgQ0yzZbg9KXKP4Jz6Yq/monitor/analysis/G9PgFG80yzZjVXz85JQQ4jP0

@commandlinegirl @Gvaihir

**Review by commandlinegirl — APPROVED** [review]
